src/verilog/aval_bval_encoding.cpp

```diff
@@ -93,18 +93,31 @@ constant_exprt lower_to_aval_bval(const constant_exprt &src)
 
 exprt aval(const exprt &src)
 {
-  PRECONDITION(is_aval_bval(src.type()));
-  auto width = aval_bval_width(src.type());
-  return extractbits_exprt{
-    src, from_integer(0, integer_typet()), bv_typet{width}};
+  if(is_aval_bval(src.type()))
+  {
+    auto width = aval_bval_width(src.type());
+    return extractbits_exprt{src, 0, bv_typet{width}};
+  }
+  else
+  {
+    auto width = to_bitvector_type(src.type()).get_width();
+    return typecast_exprt::conditional_cast(src, bv_typet{width});
+  }
 }
 
 exprt bval(const exprt &src)
 {
-  PRECONDITION(is_aval_bval(src.type()));
-  auto width = aval_bval_width(src.type());
-  return extractbits_exprt{
-    src, from_integer(width, integer_typet()), bv_typet{width}};
+  if(is_aval_bval(src.type()))
+  {
+    auto width = aval_bval_width(src.type());
+    return extractbits_exprt{src, 0, bv_typet{width}};
```

Comment:
Shouldn't this be `width` instead of `0`? (Can we have a test that catches this?)

Reply:
Fixed, test added!